networkx/algorithms/hierarchy.py

```diff
@@ -42,6 +42,9 @@ def flow_hierarchy(G, weight=None):
        DOI: 10.1002/cplx.20368
        http://web.mit.edu/~cmagee/www/documents/28-DetectingEvolvingPatterns_FlowHierarchy.pdf
     """
+    # corner case: G has no edges
+    if G.size() == 0:
+        raise nx.NetworkXError("G must have edges in flow_hierarchy")
```

Comment:
```suggestion
        raise nx.NetworkXPointlessConcept("G must have edges in flow_hierarchy")
```
Also, pls consider adding these exceptions to the docstring :)

Reply:
I think PointlessConcept is supposed to be used for the Null Graph (not empty graph). Let's talk about it in the NX community meeting -- it might be time for a change to either the name or the usage.